pkg/cmd/run/view/view.go

```diff
@@ -497,6 +499,11 @@ func logFilenameRegexp(job shared.Job, step shared.Step) *regexp.Regexp {
 	// constructs a job name by constructing a job name of `<JOB_NAME`> / <ACTION_NAME>`. This means that logs will
 	// never be found for jobs that use composite actions.
 	sanitizedJobName := strings.ReplaceAll(job.Name, "/", "")
+	// Truncate the job name to 89 characters due to zip limilation, see issue
+	// https://github.com/cli/cli/issues/7642.
+	if len(sanitizedJobName) > MAX_ZIP_FILE_PATH_LENGTH {
+		sanitizedJobName = sanitizedJobName[:MAX_ZIP_FILE_PATH_LENGTH]
```

Comment:
Could you do me a favour and create and run a workflow with some job names that include unicode chars? Something like:

```
😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅😅
```

Should be more than enough. I think this implementation is going to:
 1. Count bytes rather than chars
 2. Slice into the middle of a multi-byte char

I don't know what the underlying zip file restriction accounts for, but I'm pretty sure reading the code that the server is dealing in characters (runes in go) rather than bytes.

Reply:
This is super tricky - encoding, two-code vs three-code etc. 

Could we separate this to a different PR? For our use case English should be enough.

Reply:
At a glance I think the following snippet would match the C# behaviour but I have not validated it:

```go
func TruncateAfter(str string, max int) string {
	// Encode the string to UTF-16 to count code units
	utf16Encoded := utf16.Encode([]rune(str))
	if len(utf16Encoded) > max {
		// Decode back to UTF-8 up to the max length
		str = string(utf16.Decode(utf16Encoded[:max]))
	}
	return strings.TrimSpace(str)
}
```

This encodes the string as `UTF-16` as [used by C#](https://learn.microsoft.com/en-us/dotnet/standard/base-types/character-encoding-introduction#the-string-and-char-types), then compares the actual count of UTF-16 code units (as `str.Length` would do in C#), then slices the encoded utf-16 string and decodes that back to UTF-8.